src/rat/core/sarea/bot_filter.py

```diff
@@ -0,0 +1,236 @@
+                        curr = merged_pdf_filt3['WA_Optical_cor'].loc[:i][-1]
+                        mod_val = (merged_pdf_filt3['WA_Optical_cor'].loc[:i][-2] + merged_pdf_filt3['WA_Optical_cor'].loc[:i][-3])/2*(1+sar_trend_weekly_av)
+                        if pd.isna(mod_val):
+                            mod_val = curr
+                        merged_pdf_filt3['WA_Optical_cor'].loc[:i][-1] = mod_val
+                        merged_pdf_filt3['WA_Optical_cor_trend'] =  merged_pdf_filt3['WA_Optical_cor'].pct_change()
+                            
+                merged_pdf_filt3['WA_mean_corr'] = merged_pdf_filt3['WA_Optical_cor']
+                sa_save_path = os.path.join(tmsos_gee_path,res_name + '.csv')
+                final_sa = pd.read_csv(sa_save_path)
+                
+                final_sa['date'] = pd.to_datetime(final_sa['date'])
+                final_sa.set_index('date',inplace = True)
+                if('area_tmsos' not in final_sa.columns):
+                    final_sa['area_tmsos'] = final_sa['area']
+                final_sa['area'] = merged_pdf_filt3['WA_mean_corr']            
+                final_sa['days_passed'] = final_sa.index.to_series().diff().dt.days.fillna(0).astype(int)
+                final_sa.to_csv(os.path.join(tmsos_gee_path,res_name + '.csv'))
+                
+                botFilter_status_new = pd.DataFrame({ 'Reservoir': f'{res_name}', 'Status': 1 }, index = [curr_dam_index])
+                botFilter_status = pd.concat([botFilter_status, botFilter_status_new])
+                
+            except Exception as e:
+                log.error(f'Filtering failed for {res_name} due to error: {e}')
+                count_failed = count_failed+1
+                botFilter_status_new = pd.DataFrame({ 'Reservoir': f'{res_name}', 'Status': 0 }, index = [curr_dam_index])
+                botFilter_status = pd.concat([botFilter_status, botFilter_status_new])
+                
+        if(count_failed > 0):    
+            log_level1.info(f'BOT filter run completed with {count_failed} errors.')
+        else:
+            log_level1.info(f'BOT filter run completed succesfully')       
+        log.info('BOT Filter status\n',botFilter_status)  
+        
+    else:
+        # If Bot filter apply is set to False, reset the surface area to TMSOS.
+        for curr_dam_index,res_name in enumerate(res_list):
+            try: 
+                tmsos_gee_data = pd.read_csv(os.path.join(tmsos_gee_path,res_name + '.csv'))
+                if 'area_tmsos' in tmsos_gee_data.columns:
+                    log_level1.warning('BOT Filter toggled off. Reseting Surface Area values to TMSOS.')
+                    tmsos_gee_data['area'] = tmsos_gee_data['area_tmsos']
+            except Exception as e:
+                log_level1.error(f'BOT Filter toggled off. Reseting Surface Area values to TMSOS. \
```

Comment:
if we use log_level1.exception instead of loglevel1.error, the whole traceback is printed and you won't have to separately print e. If you just print e, only the final error will be printed that will be hard to debug.

Reply:
Removed entire check in new commit. Since BOT filter is run at end of step 10, value will be auto reset to TMSOS if false.